migrations/20180325211342_migrate-posts.js

```diff
@@ -60,6 +76,7 @@ exports.up = async knex => {
     const subArray = articleRows.slice(startIndex, startIndex + chunkSize);
     promises.push(knex('articles').insert(subArray));
     if (promises.length === 10) {
+      // eslint-disable-next-line no-await-in-loop
       await Promise.all(promises);
```

Comment:
I can imagine this pattern will be used somewhere else two. If you wanna have some fun, you could create a utility function to which you can add promises, and once they transgress a certain length, they all get executed whilst the main loop is suspended :). Nothing urgent, though.

Reply:
Yeah, for now, in the complete migrations PR I believe, I at least modularized out the batch insert as I ended up using this 3 times :)